lib/Chat/SystemMessage/Listener.php

```diff
@@ -77,281 +77,303 @@ public function __construct(IRequest $request,
+	public static function sendSystemListableMessage(ModifyRoomEvent $event): void {
+		$room = $event->getRoom();
 
-			if ($event->getNewValue() === Room::READ_ONLY) {
-				$listener->sendSystemMessage($room, 'read_only');
-			} elseif ($event->getNewValue() === Room::READ_WRITE) {
-				$listener->sendSystemMessage($room, 'read_only_off');
-			}
-		});
-		$dispatcher->addListener(Room::EVENT_AFTER_LISTABLE_SET, static function (ModifyRoomEvent $event) {
-			$room = $event->getRoom();
+		/** @var self $listener */
+		$listener = \OC::$server->get(self::class);
 
-			/** @var self $listener */
-			$listener = \OC::$server->get(self::class);
+		if ($event->getNewValue() === Room::LISTABLE_NONE) {
+			$listener->sendSystemMessage($room, 'listable_none');
+		} elseif ($event->getNewValue() === Room::LISTABLE_USERS) {
+			$listener->sendSystemMessage($room, 'listable_users');
+		} elseif ($event->getNewValue() === Room::LISTABLE_ALL) {
+			$listener->sendSystemMessage($room, 'listable_all');
+		}
+	}
 
-			if ($event->getNewValue() === Room::LISTABLE_NONE) {
-				$listener->sendSystemMessage($room, 'listable_none');
-			} elseif ($event->getNewValue() === Room::LISTABLE_USERS) {
-				$listener->sendSystemMessage($room, 'listable_users');
-			} elseif ($event->getNewValue() === Room::LISTABLE_ALL) {
-				$listener->sendSystemMessage($room, 'listable_all');
-			}
-		});
-		$dispatcher->addListener(Room::EVENT_AFTER_LOBBY_STATE_SET, static function (ModifyLobbyEvent $event) {
-			if ($event->getNewValue() === $event->getOldValue()) {
-				return;
-			}
+	public static function sendSystemLobbyMessage(ModifyLobbyEvent $event): void {
+		if ($event->getNewValue() === $event->getOldValue()) {
+			return;
+		}
 
-			$room = $event->getRoom();
+		$room = $event->getRoom();
 
-			/** @var self $listener */
-			$listener = \OC::$server->get(self::class);
+		/** @var self $listener */
+		$listener = \OC::$server->get(self::class);
 
-			if ($event->isTimerReached()) {
-				$listener->sendSystemMessage($room, 'lobby_timer_reached');
-			} elseif ($event->getNewValue() === Webinary::LOBBY_NONE) {
-				$listener->sendSystemMessage($room, 'lobby_none');
-			} elseif ($event->getNewValue() === Webinary::LOBBY_NON_MODERATORS) {
-				$listener->sendSystemMessage($room, 'lobby_non_moderators');
-			}
-		});
+		if ($event->isTimerReached()) {
+			$listener->sendSystemMessage($room, 'lobby_timer_reached');
+		} elseif ($event->getNewValue() === Webinary::LOBBY_NONE) {
+			$listener->sendSystemMessage($room, 'lobby_none');
+		} elseif ($event->getNewValue() === Webinary::LOBBY_NON_MODERATORS) {
+			$listener->sendSystemMessage($room, 'lobby_non_moderators');
+		}
+	}
 
-		$dispatcher->addListener(Room::EVENT_AFTER_USERS_ADD, static function (AddParticipantsEvent $event) {
-			$room = $event->getRoom();
-			if ($room->getType() === Room::TYPE_ONE_TO_ONE) {
-				return;
-			}
+	public static function addXJoinedTheConversationWeneverUserId(AddParticipantsEvent $event): void {
```

Comment:
Typo:
```suggestion
	public static function addXJoinedTheConversationWheneverUserId(AddParticipantsEvent $event): void {
```
But actually lets go simple:
```suggestion
	public static function addSystemMessageUserAdded(AddParticipantsEvent $event): void {
```